src/Compilers/CSharp/Test/Syntax/Parsing/ParserRegressionTests.cs

```diff
@@ -83,24 +81,7 @@ public void c024928()
         [Fact]
         public void TestBinary()
         {
-            // Apparently this fixed seed exposed a bug at some point
-            var random = new System.Random(12345);
-            // 40 million "character"s
-            const int n = 40 * 1000 * 1000;
-            var str = new string(' ', n);
-
-            unsafe
-            {
-                fixed(char* chars = str)
-                {
-                    for(int i = 0; i < n; i++)
-                    {
-                        chars[i] = (char)random.Next(char.MaxValue);
-                    }
-                }
-            }
-
-            var tree = CSharpSyntaxTree.ParseText(str);
+            CSharpSyntaxTree.ParseText(new RandomizedSourceText());
```

Comment:
We are producing parse trees for any text given as an input. Even when there are parsing/scaning errors, the parse tree is supposed produce the original text when converted to string.